src/js/components/Preview.react.js

```diff
@@ -232,6 +232,7 @@ class Preview extends React.Component<Props, State> {
   };
 
   getTabPanes = () => {
+    const qtWarnings = this.state.warnings.length;
```

Comment:
nit: how do you feel about naming the variable `warningsCount` to improve the readability?